libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java

```diff
@@ -761,6 +761,7 @@ private void cleanUp(Activity frontActivity, Account account) {
     protected void cleanUp(UserAccount userAccount) {
         SalesforceAnalyticsManager.reset(userAccount);
         RestClient.clearCaches(userAccount);
+        UserAccountManager.getInstance().clearCachedCurrentUser(userAccount);
```

Comment:
We also need to clear the cache when user switching occurs.

Reply:
Also needs to be updated on token refresh.

Reply:
We don't care about the token refresh case. The javadoc for getCachedCurrentUser() cleary states that it should only be called by methods that only care about the identity of the current user (org id, user id) and that oauth tokens might be outdated.

I'll look at user switching.

Reply:
True, but it's low cost to update this when the token changes - an intent is broadcast. If we include that, this cached account should always be in sync and can be used in more places.

Reply:
If we do this, do we need a new API? Can we simply use the existing current user API while it maintains the cached value internally in an instance variable?